autogpt_platform/frontend/src/lib/autogpt-server-api/client.ts

```diff
@@ -762,6 +762,10 @@ export default class BackendAPI {
     await this._request("DELETE", `/library/presets/${presetID}`);
   }
 
+  getAyrshareSSOUrl(): Promise<{ sso_url: string; expire_at: string }> {
+    return this._get("/integrations/ayrshare/sso_url");
+  }
```

Comment:
I prefer that we use the new generated queries rather than keep building on the old `BackendAPI` client. If this endpoint is on OpenAPI, then running `pnpm generate:api-all` then you should get the hook and types generated for you under `src/api/__generated__/endoints/integrations` 👍🏽 

Explanation here on how to use them:
https://github.com/Significant-Gravitas/AutoGPT/blob/master/autogpt_platform/frontend/README.md#queries-get-requests

So we won't need the `useState` above, given react-query manages the loading and error states for us 💆🏽

Reply:
update to use the new method, thanks